### PR TITLE
Fixes on SensorThings

### PR DESCRIPTION
**Pull request by tcalmant**
* Use internal GeoJSON library
* Fixed filters behaviour when related argument is missing
* Don't use `@Context` to find the registered `ObjectMapper` and `SensinactSession` providers
* Normalized conversion of `admin/location` to a `GeoJsonObject`

### CODE REVIEW

File: prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/OrderByFilter.java [hunk 1]
```diff
@@ -43,17 +45,26 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
...
+            List<Self> sortedList = new ArrayList<>(resultList.value);
+            sortedList.sort(comparator);
+            resultList.value = List.copyOf(sortedList);
         }
     }
 
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
         List<String> list = requestContext.getUriInfo().getQueryParameters().get("$orderby");
+        if (list == null) {
```
Comment (timothyjward) [inline prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/OrderByFilter.java:65]: This behaviour is wrong. In the absence of a sort property we should default to sorting by id as per the spec

File: prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/OrderByFilter.java [hunk 2]
```diff
@@ -43,17 +45,26 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             throws IOException {
         @SuppressWarnings("unchecked")
         Comparator<Object> comparator = (Comparator<Object>) requestContext.getProperty(ORDERBY_PROP);
+        if (comparator == null) {
```
Comment (timothyjward) [inline prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/OrderByFilter.java:48]: There should always be a comparator set to sort the results.

File: prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/CountFilter.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             throws IOException {
         Boolean addCount = (Boolean) requestContext.getProperty(COUNT_PROP);
 
-        if(addCount) {
+        if(addCount != null && addCount) {
```
Comment (timothyjward) [inline prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/CountFilter.java:43]: Add count should not be null, see the bug below

File: prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/PropFilterImpl.java [hunk 4]
```diff
@@ -38,6 +38,12 @@ public void aroundWriteTo(WriterInterceptorContext context) throws IOException,
         Object entity = context.getEntity();
 
         String propName = uriInfo.getPathParameters().getFirst("prop");
+        if(propName == null) {
+            // No property filter
```
Comment (timothyjward) [inline prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/PropFilterImpl.java:42]: Isn’t this an error? The prop filter name binding is supposed to be used with a path parameter that cannot be null, so I’m not sure how this code is reached. If it is reached we should abort with a failure.

Comment (tcalmant) [reply prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/PropFilterImpl.java:42]: Agreed. This case only happens if the annotated method doesn't have a "prop" parameter.

**Pull request merged**